domain/keymanager/state/types.go

```diff
@@ -25,7 +27,33 @@ type publicKeyData struct {
 	PublicKey string `db:"public_key"`
 }
 
-// userId represents a user id for associating public keys with.
-type userId struct {
-	UserId string `db:"user_id"`
+// userPublicKeyId represents a single raw user public key id from the database.
+type userPublicKeyId struct {
+	Id int64 `db:"id"`
+}
+
+// userPublicKeyIds represents an aggregate slice of [userPublicKeyId] for
+// performing bulk in operations.
+type userPublicKeyIds []userPublicKeyId
+
+// userUUIDValue represents a user id for associating public keys with.
+type userUUIDValue struct {
+	UUID string `db:"user_uuid"`
+}
+
+// modelUUIDValue represents a model id for associating public keys with.
+type modelUUIDValue struct {
+	UUID string `db:"model_uuid"`
+}
+
+// modelAuthorizedKey represents a single row from the model_authorized_keys
+// table.
+type modelAuthorizedKey struct {
+	UserPublicSSHKeyId int64  `db:"user_public_ssh_key_id"`
+	ModelUUID          string `db:"model_uuid"`
+}
+
+// Value returns the user id implementing the [driver.Valuer] interface.
+func (u userPublicKeyId) Value() (driver.Value, error) {
```

Comment:
Move to line 35?

Reply:
I have always defined types at the top of the file and then funcs after. This has been a pattern in most of our code. Some times we seem to deviate off of this but it is the norm from what I can tell. Do we have a decision around this somewhere?

Reply:
Not a big thing, but all userPublicKeyId stuff should be put together.